packages/swap/src/quoting/PoolStateCache.ts

```diff
@@ -11,28 +17,34 @@ const rpcConfig = { rpcUrl: env.RPC_NODE_HTTP_URL };
 
 type BaseAsset = Exclude<InternalAsset, 'Usdc'>;
 
-const baseAssets = (Object.keys(InternalAssets) as InternalAsset[]).filter(
-  (asset): asset is BaseAsset => asset !== 'Usdc',
-);
+let baseAssets: BaseAsset[];
 
 type PoolState = {
   poolState: string;
   rangeOrderPrice: bigint;
 };
 
-const fetchPoolState = async (hash: string) =>
-  Object.fromEntries(
+const fetchPoolState = async (hash: string) => {
+  baseAssets ??= (await getSupportedAssets(rpcConfig))
+    .filter((asset) => asset.chain !== 'Ethereum' || asset.asset !== 'USDC')
```

Comment:
```suggestion
    .filter((asset) => !(asset.chain === 'Ethereum' && asset.asset === 'USDC'))
```
just a preference, not sure if it's easier to read for others as well or not

Reply:
definitely easier to read